### PR TITLE
Stop serving when leaving project

### LINKED ISSUE #74: Stop Serving when leaving project
When the user leave the current project, the client should immediately stop serving the current project before leaving

### PR DESCRIPTION
**Pull request by ZimGil**
# Stop serving when leaving project

 - stopServing() command in leaveProject() [resolves #74]
   > If an `ng serve` command is running, will send a stop command
 
 - `stopServing()` moved to `app.component`

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -91,11 +92,35 @@ export class AppComponent implements OnInit {
           return this.commandService.getProjects();
       })
     )
-    .subscribe((projects: string[]) => {
+    .pipe(tap((projects: string[]) => {
```
Comment (ortichon) [inline src/app/app.component.ts]: you don't need the pipe

File: src/app/app.component.ts [hunk 2]
```diff
@@ -118,13 +143,22 @@ export class AppComponent implements OnInit {
     const request = new CommandRequest(userCommand);
 
     this.commandService.sendCommand(request).pipe(
-      mergeMap(commandId => timer(0, this.COMMAND_STATUS_CHECK_INTERVAL)
+      mergeMap(commandId => {
```
Comment (ortichon) [inline src/app/app.component.ts]: rethink this one, please. the server should return `error` status and then you'll just need to add `error` case down below

**Comment by ZimGil** [comment]
The server will now keep the current `ngServe` command and will be able to notify the client upon `http.get` if the're an `ngServe` command running.

File: src/app/app.component.ts [hunk 3]
```diff
@@ -114,51 +114,34 @@ export class AppComponent implements OnInit {
         });
       },
       () => {
-        this.serveCommandId = null;
-        localStorage.removeItem('ngServeCommandId');
+        this.isServing = false;
```
Comment (ortichon) [inline src/app/app.component.ts]: remove finally, move logic to the first callback

**Pull request merged**